File: 2022-2023/sourcecode/lab/lab12/InterfaceInjectBasicsJava/app/src/main/java/pl/edu/uwr/pum/interfaceinjectbasicsjava/di/modules/IntelModule.java
package pl.edu.uwr.pum.interfaceinjectbasicsjava.di.modules;

import dagger.Module;
import dagger.Provides;
import pl.edu.uwr.pum.interfaceinjectbasicsjava.computer.component.AMD;
import pl.edu.uwr.pum.interfaceinjectbasicsjava.computer.component.CPU;
import pl.edu.uwr.pum.interfaceinjectbasicsjava.computer.component.Intel;

@Module
public class IntelModule {
    @Provides
    CPU provideIntel(Intel cpu){
        return cpu;
    }
}

File: 2022-2023/sourcecode/lab/lab12/InterfaceInjectiBasicsKotlin/app/src/main/java/pl/edu/uwr/pum/interfaceinjectibasicskotlin/di/modules/IntelModule.kt
package pl.edu.uwr.pum.interfaceinjectibasicskotlin.di.modules

import dagger.Module
import dagger.Provides
import pl.edu.uwr.pum.interfaceinjectibasicskotlin.computer.components.CPU
import pl.edu.uwr.pum.interfaceinjectibasicskotlin.computer.components.Intel

@Module
class IntelModule {
    @Provides
    fun provideCpu(cpu: Intel): CPU {
        return cpu
    }
}